Fix visibility of various C++ functions, inspired by a similar patch in the Fink
project: 
http://www.mail-archive.com/fink-commits@lists.sourceforge.net/msg75742.html

JL 05/07/10 
Index: beecrypt-4.1.2/c++/security/Security.h
===================================================================
--- beecrypt-4.1.2.orig/c++/security/Security.h	2010-07-05 17:15:34.811273306 +0100
+++ beecrypt-4.1.2/c++/security/Security.h	2010-07-05 17:15:38.696388531 +0100
@@ -75,8 +75,6 @@
 		public:
 			typedef vector<const Provider*> provider_vector;
 			typedef provider_vector::iterator provider_vector_iterator;
-
-		private:
 			struct spi
 			{
 				Object* cspi;
@@ -91,6 +89,7 @@
 			static spi* getSpi(const String& algo, const String& type, const Provider&) throw (NoSuchAlgorithmException);
 			static spi* getFirstSpi(const String& type);
 
+		private:
 			static const String& getKeyStoreDefault();
 
 			static bool _init;
Index: beecrypt-4.1.2/c++/beeyond/BeeCertificate.h
===================================================================
--- beecrypt-4.1.2.orig/c++/beeyond/BeeCertificate.h	2004-11-03 09:37:27.000000000 +0000
+++ beecrypt-4.1.2/c++/beeyond/BeeCertificate.h	2010-07-05 17:16:25.216332324 +0100
@@ -150,11 +150,11 @@
 			mutable String* str;
 
 			BeeCertificate();
-			BeeCertificate(InputStream& in) throw (IOException);
 
 			bytearray* encodeTBS() const;
 
 		public:
+			BeeCertificate(InputStream& in) throw (IOException);
 			BeeCertificate(const BeeCertificate&);
 			virtual ~BeeCertificate();
 
